pkg/testutils/serverutils/test_server_shim.go

```diff
@@ -108,6 +108,14 @@ func ShouldStartDefaultTestTenant(
 		return baseArg
 	}
 
+	if multiNodeCluster && util.RaceEnabled {
+		// Race builds now run in the EngFlow environment which seems to be
+		// often overloaded if we have multi-node clusters and start default
+		// test tenant, so we disable the tenant randomization in such a
+		// scenario.
+		return base.InternalNonDefaultDecision(baseArg, false /* enable */, false /* shared */)
+	}
```

Comment:
Could we put this under the `testTenantDecisionFromEnvironment` block so it is possible to run a test under race with tenants enabled locally still?